File: Pythonleetcode/Leetcode11.py
class Solution(object):
    def maxArea(self, height):
        max_area = 0

        left = 0

        right = len(height) - 1

        while right > left:

            max_area = max(max_area, min(height[left], height[right]) * (right - left))

            if height[right] > height[left]:
                left += 1
            else:
                right -= 1
        return max_area





height=[2,1,3,4,1,5,2]

print(Solution().maxArea(height))

File: CppPractice/Project1/Project1/Leetcode11.cpp
#include <string>

#include <vector>

#include <iostream>

#include <stdlib.h>

#include <time.h>

#include <sstream>

#include <algorithm>

#include <map>

#include <bitset>

using namespace std;

class Solution {
public:
    int maxArea(vector<int>& height) {
        int ans = 0;
        int left = 0;
        int right = height.size() - 1;
        while (left<right)
        {
            ans = max(ans, (right - left) * min(height[left], height[right]));
            if (height[left]<height[right])
            {
                left++;
            }
            else
            {
                right--;
            }

        }
        return ans;

    }
};